### PR TITLE
[RFR] Failsafe and verbose

### PR DESCRIPTION
**Pull request by StefanRijnhart**
Tentative patch that I'm now running the migration with.

### CODE REVIEW

File: queue_job/migrations/10.0.1.0.0/end-migration.py [hunk 1]
```diff
@@ -31,6 +32,18 @@ def migrate(cr, version):
...
+    error = 0
+    for record in records:
+        count += 1
+        if not count % 1000:
+            logger.info("Recomputing func_string of job %s of %s",
+                        count, total)
+        try:
+            record._compute_func_string()
+        except KeyError:  # unknown model or non-compliant arguments
+            error += 1
```
Comment (hbrunn) [inline queue_job/migrations/10.0.1.0.0/end-migration.py:47]: please log the error so that I see what's going wrong here

Comment (StefanRijnhart) [reply queue_job/migrations/10.0.1.0.0/end-migration.py:47]: It's this one https://github.com/OCA/queue/pull/341#discussion_r636695433 but I've pushed a logging commit. Restarting the migration now with this code.

**Comment by StefanRijnhart** [comment]
Applying the operation on all 200000 jobs takes about 6 minutes, so that is alright. I made a big hooha about operations on larger amounts of jobs, but removing the jobrunner trigger from the table solved that later on.

File: queue_job/migrations/10.0.1.0.0/end-migration.py (hunk 1 above)
Comment (StefanRijnhart) [reply queue_job/migrations/10.0.1.0.0/end-migration.py:47]: I'm attaching the relevant logging to the ticket.

**Comment by StefanRijnhart** [comment]
Replaced deprecated logger.warn by logger.warning

**Comment by StefanRijnhart** [comment]
Can you please complete your review? Or are you not considering merging this at all?

**Review by hbrunn — APPROVED** [review]
I thought this was meant just for debugging as you PRed on the private repo, but this will probably be useful for other people too.

I still don't really get it why we care for done jobs at all, but then I don't have much experience with things you might want to do with old jobs.

**Comment by StefanRijnhart** [comment]
We care for done jobs because of code like https://github.com/VanMoof/odoo-addons/blob/12.0/vanmoof_magento/models/magento_sale_order.py#L27-L34.

I will be proposing to override this code again with a custom change that defers recomputing func_string here. So for general usefulness I would change this PR to remove the logging of the tracebacks on every KeyError, and if you want I can restore the filter on non-done jobs. Please let me know if that has your preference.

**Comment by hbrunn** [comment]
ah, I see. No worries about non-done jobs, I've no claims here

**Comment by StefanRijnhart** [comment]
Removed exc_info=True from the debug log, so whenever you are ready you can cherry-pick or rebase this to make it show up in your public OCA PR.

**Comment by hbrunn** [comment]
thanks, as it's now a part of https://github.com/OCA/queue/pull/341 I'll close this

**Pull request closed without merging**